app/models/user.rb

```diff
@@ -106,6 +106,50 @@ def gpo_verification_pending_profile?
     gpo_verification_pending_profile.present?
   end
 
+  def suspended?
+    suspended_at.present?
+  end
+
+  def reinstated?
+    reinstated_at.present?
+  end
+
+  def suspend
+    confirm_that_user_can_be_suspended!
+    update!(suspended_at: Time.zone.now)
+    track_user_suspended(success: true)
+  end
+
+  def reinstate
+    confirm_that_user_can_be_reinstated!
+    update!(reinstated_at: Time.zone.now)
+    track_user_reinstated(success: true)
+  end
+
+  def confirm_that_user_can_be_suspended!
+    if reinstated?
+      error_message = 'Cannot suspend a user who has already been reinstated.'
```

Comment:
I am not sure this is true. It's uncommon, but from talking to Team Judy it's possible an account would look fraud-ish, but not egregiously so, and get reinstated, and then later turn out to be fraudulent. I think you should be able to suspend a reinstated account.